src/main/java/com/github/magickoders/JarReader.java

```diff
@@ -0,0 +1,153 @@
+package com.github.magickoders;
+
+import com.github.forax.umldoc.core.Entity;
+import com.github.forax.umldoc.core.Modifier;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Opcodes;
+
+import java.io.IOException;
+import java.lang.module.ModuleFinder;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * A class used to retrieve the content of a jar.
+ * It contains a static method {@link #getEntities()} which returns the entities contained in the directory "target"
+ */
+public class JarReader {
+
+    /**
+     * A class that contains the list of entities by using the ClassVisitor's methods
+     */
+    private static class MyVisitor extends ClassVisitor { // bwaaa de l'héritage, mais c'est une class abstraite donc ok j'imagine
+
+        private final List<Entity> entities = new ArrayList<>();
+
+
+        public List<Entity> getEntities() {
+            return entities;
+        }
+
+        protected MyVisitor(int api) {
```

Comment:
protected is useless, given that the class is private, the constructor should be private